localstack/utils/aws/templating.py

```diff
@@ -135,9 +135,7 @@ class ExtAssignment(airspeed.operators.Assignment):
     Extends the airspeed Assignment class to support names with dashes, e.g., "X-Amz-Target"
     """
 
-    START = re.compile(
-        r"\s*\(\s*\$([a-z_][a-z0-9_]*(?:\.[a-z_][a-z0-9_-]*)*)\s*=\s*(.*)$", re.S + re.I
-    )
+    START = re.compile(r"\s*\(\s*\$(\w*(?:\.[\w-]+|\[\"\$\w+\"\]*)*)\s*=\s*(.*)$", re.S + re.I)
```

Comment:
Would be great to get these patches added to our airspeed fork directly at some point.

Question - seems like the semantics of the first part of the regex has slightly changed, should we have `\w+` instead of `\w*`?
```suggestion
    START = re.compile(r"\s*\(\s*\$(\w+(?:\.[\w-]+|\[\"\$\w+\"\]*)*)\s*=\s*(.*)$", re.S + re.I)
```

Reply:
Yep, the goal is to migrate from here to airspeed, it makes sense.

For correctness, we need after a `$` one or more characters as a variables, hence the `+`. But I think this is harmless so will keep it as before.